### DISCUSSION TITLE
DS18B20 sensors as W1Device

### DISCUSSION THREAD

**Discussion by RobinQuante** [Pi4J/pi4j#153] [Q&A]
Hello,

i have 2 DS18B20 sensors with the one wire communication.
i have read all devices and geht back the ID  28-3c01e0769c2e
but when i call 
master.getDeviceById("28-3c01e0769c2e");
the master dont find any Device and return null.

**Comment by savageautomate** [comment]
This looks like Pi4J v1.x code.

**Reply by mateusz1364** [reply]
Is any plan to add One Wire communication to Pi4J 2.x ?

**Comment by FDelporte** [comment]
No component-specific implementation is foreseen, but I think the DHT11 sensor is also a one-wire device, and an example implementation is available in the CrowPi project:

https://github.com/Pi4J/pi4j-example-crowpi/blob/main/src/main/java/com/pi4j/crowpi/components/HumiTempComponent.java

**Reply by mateusz1364** [reply]
This example using retrieve measurement values from file in system storage and this is not a good solution.
In my opinion in Pi4J should will be driver to interacting with one wire devices directly.

**Reply by FDelporte** [reply]
as this is an open-source project, we welcome any improvement of the library! please feel free to create a ticket and contribute a first draft so we can guide you in the process

**Reply by mateusz1364** [reply]
Ok, and this is a challenge :) When I do sth, I will share it. Thanks!

**Comment by msseodev** [comment]
Pi4J v2 still does not support 1-wire(like DS18B20)?
I am finding a solution but only I could find was using pi4j V1.

**Reply by taartspi** [reply]
https://github.com/Pi4J/pi4j-example-devices    There is a dht22 implementation using a gpio for the one wire communication.  This is a more simple chip than yours as it requires one toggle of the pin for the reading operation to begin where your chip supports a great deal of configuration driven from the Pi.  But perhaps this will assist your design of the low level code for your chip. 

your actual question. No one wire support has been added to Pi4j v2
